sambacc/commands/ctdb.py

```diff
@@ -366,3 +367,47 @@ def ctdb_must_have_node(ctx: Context) -> None:
         ctdb.cluster_meta_to_nodes(
             np.cluster_meta(), real_path=np.persistent_path
         )
+
+
+def _ctdb_rados_mutex_args(parser: argparse.ArgumentParser) -> None:
+    parser.add_argument(
+        "--cluster-name",
+        default="ceph",
+        help="Cluster name to pass to mutex lock helper",
+    )
+    parser.add_argument(
+        "mutex_uri",
```

Comment:
https://docs.python.org/3/library/argparse.html#name-or-flags
This is meant to be a positional argument, not an optional argument. A quirk of argparse is that optional arguments will automatically be converted to python var names (for the `dest` if not supplied) by converting (non-leading) dashes to underscores. But a positional argument doesn't.

So in the ctdb.conf the command is specified like so:
`cluster lock = !/usr/bin/samba-container ctdb-rados-mutex rados://.smb/ns/cluster.meta.cluster_lock`  as in https://github.com/phlogistonjohn/ceph/blob/dfb9f9fcdb7aaa17f5ff8e5f1d06fd4291d02bbc/src/cephadm/cephadmlib/daemons/smb.py#L627
as the cluster-id value is sourced from the environment and the cluster-name is left to use the default because I have no idea if this value has any practical use.

Reply:
> https://docs.python.org/3/library/argparse.html#name-or-flags This is meant to be a positional argument, not an optional argument. A quirk of argparse is that optional arguments will automatically be converted to python var names (for the `dest` if not supplied) by converting (non-leading) dashes to underscores. But a positional argument doesn't.

Aha..I better read something before making a comment, especially with Python 😆.

> So in the ctdb.conf the command is specified like so: `cluster lock = !/usr/bin/samba-container ctdb-rados-mutex rados://.smb/ns/cluster.meta.cluster_lock` as in https://github.com/phlogistonjohn/ceph/blob/dfb9f9fcdb7aaa17f5ff8e5f1d06fd4291d02bbc/src/cephadm/cephadmlib/daemons/smb.py#L627 as the cluster-id value is sourced from the environment and the cluster-name is left to use the default because I have no idea if this value has any practical use.

I hope you meant ceph_id(and not cluster-id) and I see the place where you set `SAMBACC_CEPH_ID`.

Perfect, this format looks much much better than what I thought of.